.github/workflows/nx-cloud-main.yml

```diff
@@ -42,6 +45,7 @@ on:
 
 env:
   NX_CLOUD_DISTRIBUTED_EXECUTION: true
+  NX_CLOUD_DISTRIBUTED_EXECUTION_AGENT_COUNT: ${{ inputs.number-of-agents || 6 }}
```

Comment:
Why 6? Our docs use 3 everywhere so I imagine that is what most existing users have copy pasted.

I didn't know about this flag until this PR, and the docs are not very good: https://nx.dev/nx-cloud/reference/env-vars#environment-variables

So can we also use this moment to improve those?

E.g. what if this number is not present vs what if it does not match the agent count vs what if it is present and perfectly aligned - would be good to outline the different scenarios.

Because, for example, what you are setting up here is for every workspace already using this to set it to 6 (because none of them will have the input), but I imagine a tiny fraction are actually using 6, so is that better or worse than no setting at all?

Maybe making it a required input is preferable as it might help cut down on the cases where the numbers do not match? It would still be possible of course but hopefully much more transparent to the user. Maybe the cloud client could (and maybe UI) could show a warning when it realizes it has fewer/more agents than it was told to expect?

I will forward these comments on to Victor as well

Reply:
I will add more context to the PR's description.

Reply:
I agree that docs should be improved. I have that in WIP. PR will soon follow.

I wouldn't make it mandatory, simply because NxCloud can operate without it and it would be breaking change.

Reply:
FYI: https://github.com/nrwl/nx/pull/10812

Reply:
if I understood this conversation correctly, I agree with James, should we skip providing a default? i.e. if someone doesn't provide it, keep it undefined (rather than defaulting to 6). That way, we give NxCloud a chance to infer the number, versus assuming it's always 6.

Reply:
Nx Cloud *doesn't* infer the number. We will just batch tasks as if there were 6 agents if there is no value specified. There is nothing to try and guess the number of agents based on how much polling we are getting for a given run group, for example.

Reply:
that makes sense! the reason I thought we should keep it undefined is then the logic for deciding how to batch is in one place - i.e. if we ever decide to batch as if there were 10 agents (instead of 6), we would only have to change that on the nx-api side (rather than here as well)

Reply:
I guess leaving it `undefined` doesn't do harm (in contrast to setting it to 6)?

Can you check the text on README.md to make sure it makes sense?